.github/workflows/deploy.yml

```diff
@@ -56,25 +61,26 @@ jobs:
         # since we didn't setup a remote backend to store state,
         # following runs will fail with name conflict, which is normal.
         continue-on-error: true
-        
+
         run: |
           terraform init
           terraform apply -auto-approve -input=false
         env:
           CLOUDFLARE_API_TOKEN: ${{ secrets.CLOUDFLARE_API_TOKEN }}
-          TF_VAR_CLOUDFLARE_ACCOUNT_ID: ${{ steps.fetch_account_id.outputs.account_id }}
+          TF_VAR_CLOUDFLARE_ACCOUNT_ID: ${{ secrets.CLOUDFLARE_ACCOUNT_ID || steps.fetch_account_id.outputs.account_id }}
```

Comment:
Maybe `secrets.CLOUDFLARE_ACCOUNT_ID ||` is not neccessary here. You have set `fetch_account_id.outputs.account_id` to `secrets.CLOUDFLARE_ACCOUNT_ID` above?